### ISSUE TITLE
Feature request: tagged template raw queries

### ISSUE THREAD

**Issue by KristjanTammekivi** [koskimas/kysely#60]
Hi, looking good so far, but this is a feature that I would like to see, something like this
```javascript
await db.sql<User & { posts: Post[] }>`
SELECT
    u.*,
    (SELECT json_agg(post) FROM post WHERE user_id = user.id) as posts
FROM user
WHERE id = ${ userId }`
```
and userId would be escaped automatically. When doing raw queries with bindings as an array at the end it's quite difficult to keep track on which ? is should correspond to which binding.

**Comment by koskimas** [comment]
Sorry about the delay.

I've thought about this too and I think it's a good idea. Maybe we could even replace the `raw` function with this. I'll see how this fits in.

**Comment by koskimas** [comment]
Everything seems cleaner and better with tagged templates 😄 

# Completely raw query

```ts
await db.raw('select * from person where id = ?', [id])
```

```ts
await db.sql`select * from person where id = ${id}`
```

# Subquery

```ts
await db
  .selectFrom('person')
  .select(
    db.raw<number>(`(
      select count(pet.id) 
      from pet
      where species = ?
      and pet.owner_id = person.id
    )`, [species]).as('pet_count')
  )
```

```ts
await db
  .selectFrom('person')
  .select(
    db.sql<number>`(
      select count(pet.id) 
      from pet
      where species = ${species}
      and pet.owner_id = person.id
    )`.as('pet_count')
  )
```

I think we are going to replace `raw` with the `sql` template you suggested in the next version.

**Comment by KristjanTammekivi** [comment]
Something to keep in mind is that most tagged template sql libraries also ship helpers for cases like when you need to dynamically provide columns to select or fields to update, like postgres.js does here
https://github.com/porsager/postgres#dynamic-query-helpers---sql-inside-tagged-template

or slonik here
https://github.com/gajus/slonik#sqlidentifier

I didn't want to break anyone's usecases, that's why I suggested a separate method.

**Comment by koskimas** [comment]
Yep, I'll add a way to use identifiers.

> I didn't want to break anyone's usecases, that's why I suggested a separate method.

I don't want to have two functions that do the same thing. Kysely doesn't yet have that many users. Now's the time to break the API if it improves it.

**Comment by koskimas** [comment]
This is now implemented in [0.17.0](https://github.com/koskimas/kysely/releases/tag/0.17.0)
